src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/Target_X64/DelayLoadHelperThunk.cs

```diff
@@ -16,8 +16,20 @@ public partial class DelayLoadHelperThunk
     {
         protected override void EmitCode(NodeFactory factory, ref X64.X64Emitter instructionEncoder, bool relocsOnly)
         {
-            // lea rax, [pCell]
-            instructionEncoder.EmitLEAQ(X64.Register.RAX, _instanceCell);
+            if (_isVirtualStubDispatchCell)
+            {
+                // lea rax, r11 - this is the most general case as the value of R11 also propagates
+                // to the new method after the indirection cell has been updated so the cell content
+                // can be repeatedly modified as needed during virtual / interface dispatch.
+                instructionEncoder.EmitMOV(X64.Register.RAX, X64.Register.R11);
```

Comment:
Nit: I would add `using ILCompiler.DependencyAnalysis.X64` at the top. This file is X64 only, so we don't need to qualify what kind of `Register` this is.